library/rendering/item_rendering/src/testmod/java/org/quiltmc/qsl/rendering/item/test/client/ItemDecorations.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.rendering.item.test.client;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.BufferBuilder;
+import com.mojang.blaze3d.vertex.Tessellator;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.rendering.item.api.client.GuiRendererHelper;
+import org.quiltmc.qsl.rendering.item.test.ItemRenderingTestmod;
+
+@Environment(EnvType.CLIENT)
+public class ItemDecorations {
+	private ItemDecorations() { }
+
+	public static final Identifier STACK_BORDER_TEXTURE = ItemRenderingTestmod.id("textures/gui/border.png");
```

Comment:
Hm. Are these texture stitched into the item texture Atlas? If not, this may cause some performance issues if a lot of mods follow this trend.

Reply:
Texture atlases are extremely complex, and the GUI texture usually used is not atlased. This would be quite an invasive change to Minecraft if this were to be fixed. Not sure the best way to fix it though.